hesinfo.1
.\" Copyright 1987 by the Massachusetts Insitute of Technology.
.\" The file <mit-copyright.h> gives the terms of distribution.
.\"
.\"	$Source: /afs/dev.mit.edu/source/repository/athena/lib/hesiod/hesinfo.1,v $
.\"	$Author: probe $
.\"	$Athena: hesinfo.1,v 1.3 88/08/07 21:52:12 treese Locked $
.\"	$Header: /afs/dev.mit.edu/source/repository/athena/lib/hesiod/hesinfo.1,v 1.5 1989-12-03 18:04:04 probe Exp $
.\"	$Log: not supported by cvs2svn $
.\"	Revision 1.4  88/08/07  23:17:48  treese
.\"	Second-public-distribution
.\"
.\"	Revision 1.3  88/08/07  21:52:12  treese
.\"	First public distribution
.\"	
.\"	Revision 1.2  88/06/05  19:51:12  treese
.\"	Cleaned up for public distribution
.\"
.TH HESINFO 1 "13 November 1987"
.FM mit
.SH NAME
hesinfo \- find out what is stored in the Hesiod database
.SH SYNOPSIS
.nf
hesinfo \fI[-bl]\fR \fIHesiodName\fR \fIHesiodNameType\fR
.SH DESCRIPTION
.I hesinfo
takes two arguments, a name to be resolved and a string, known
as a HesiodNameType.  It then prints the information returned by
the Hesiod nameserver.
.PP
The value returned by
.B hesinfo
is of the type
.I HesiodNameType.
.PP
.I hesinfo
understands the following options:
.TP
.B \-l
Selects long format.
.TP
.B \-b
Prints the fully-qualified string passed to the nameserver.

.PP
.SH VALID Hesiod_Names
The following types of identifiers may be used in the
.I HesiodName
argument to
.B hesinfo.
These values will be resolved by accessing the
.B hesiod
database.

.TP
.B \<username>
the 8-character-or-less string used to identify users or classes
(e.g. joeuser, root, 1.00, etc).
Used with the
.I Hesiod_Name_Types
.BR passwd,
.BR pobox,
and
.BR filsys.


.TP
.B \<uid>
the id number assigned to a user.

.TP
.B \<groupid>
the id number assigned to a group.

.TP
.B \<groupname>
a name identifying a unique group.

.TP 
.B \<file-system-name>
the name of an athena file system.

.TP
.B \<rvd server>:<pack>
the name of an rvd's server and pack  seperated by a colon.

.TP
.B \<nfs server>:<partition>
the name of an nfs server and its partition seperated by a colon.

.TP
.B \<workstation-name>
the machine name of an Athena workstation (e.g. E40-343-3).

.TP 
.B \<service-name>
name of an Athena service (e.g. Kerberos or Zephyr).

.TP
.B \<service-type>
name of Unix service (valid entries are defined in /etc/services).

.TP
.B \<printer-name>
name of a printer.

.TP
.B \<printer-cluster-name>
name of an Athena print cluster.

.TP
.B \<foo>
some
.B hesinfo
calls (e.g. 
.B prclusterlist
) do not require a specific
.I HesiodName
argument.  However, you must include a dummy string (e.g. 'foo') for
.B hesinfo
 to work properly.




.PP
.SH VALID Hesiod_Name_Types
The following symbols are valid substitutions for the
.I HesiodNameType
argument to
.B hesinfo.

.TP
.B \ passwd
returns string suitable for inclusion in
.I /etc/passwd,
searching with
.B <username>.

.TP
.B \ pobox
returns information on the pobox assigned to the user specified by
.I HesiodName,
searching with
.B <username>.


.TP
.B \ uid   
returns string suitable for inclusion in
.I /etc/passwd,
searching with
.B <uid>.

.TP
.B \ gid   
returns string suitable for inclusion in
.I /etc/group,
searching with
.B <groupid>.

.TP
.B \ group
returns string suitable for inclusion in
.I /etc/group,
searching with
.B <groupname>.

.TP
.B \ grplist
returns subgroups included in superset
defined by 
.B <groupname>.

.TP
.B \  filsys
returns file system type, export point, server, mount mode, and import point
for the following valid
.I HesiodNames
(see above)
.B - <file system name>, <username>, <rvd server>:<pack>, 
.B and <nfs server>:<partition>

.TP
.B \ cluster
returns information about the local cluster the workstation, specified by
.B <workstation name>.
Included is information about the local file and print servers.  This
information is accesses by 
.B clusterinfo 
at boot time.

.TP
.B \ sloc
returns network name of service host for
.B <service-name>.

.TP
.B \ service
returns Internet protocol type and protocol service port for
.B <service-type>.

.TP
.B \ pcap
returns a valid entry for /etc/printcap for
.B <printer-name>.

.TP 
.B \ prcluserlist
returns a list of print clusters.

.TP 
.B \ prcluster
returns a list of printers in a cluster specified by
.B <printer-cluster-name>.

.SH FILES
/etc/hesiod.conf
.SH "SEE ALSO"
`Hesiod - Project Athena Technical Plan -- Name Service', named(8), hesiod(3)
.SH AUTHOR
Steve Dyer, IBM/Project Athena
.br
Copyright 1987, 1988 by the Massachusetts Institute of Technology.
.br
.SH BUGS


NEWS
Major changes in release 3.2.0:

* Entirely rewrite the buildsystem to use automake.
* Incorporate 3 patches from Fedora, fixing issues when libhesiod was
  used from setuid programs, the response from the DNS server was
  longer than 1024 bytes, and in which string lengths were ignored
  while doing a case-insensitive comparison.
* Ship with Debian packaging.

Major changes in release 3.1.1:
* If available, libhesiod will use libidn to internationalize a domain
  name before returning it from hesiod_to_bind()
* LDFLAGS as set by autoconf aren't ignored when building hestest or
  libhesiod

Major changes in release 3.1.0:
* hesinfo has been moved into a separate source tree.
* The build system now installs a shared library by default.
* A new function, hesiod_free_string, ensures that strings can be
  freed at the same library level as they are allocated.  (Important
  in some runtime environments.)
* A new function, hesiod_parse_result, can be used in combination with
  hesiod_to_bind to allow Hesiod resolutions to be done using an
  alternate resolver.
* Support for class HS is removed; Hesiod resolution is always done in
  class IN now.

Major changes in release 3.0.2:
* Better observance of Autoconf environment variable conventions (you
  can set CPPFLAGS, CFLAGS, LDFLAGS, and LIBS).  CDEBUGFLAGS is gone;
  CFLAGS specifies debugging and optimization flags only.
* Bugs fixed: "make clean" missed config.h, hesiod_to_bind() would
  core dump on names with right hand sides, hesiod_to_bind() didn't
  length-check its arguments.

Major changes in release 3.0.1:
* hes_init() function added to backward compatibility interfaces.
* You can use "#ifdef HESIOD_INTERFACES" in an application to test if
  the new interfaces are available.

Major changes in release 3.0.0:
* This NEWS file added.
* Wire protocol change: Hesiod data on the name servers should now be
  in class C_IN, not C_HS.  Data in class C_HS will still work for
  now, but this backward compatibility hack may not stick around
  forever.  (You can specify which classes to look at in your
  hesiod.conf file.)
* Interface change: the preferred interfaces are now named
  hesiod_*() and take a context argument.  They also return allocated
  memory which must be freed by the caller.  The *_r() interfaces are
  gone, and the hes_*() interfaces are deprecated.
* Hesiod no longer looks for /etc/hesiod.conf (unless sysconfdir is
  /etc).  It does, however, allow the environment variable
  HESIOD_CONFIG to specify where the configuration file lives.
* hesiod.conf is no longer case-sensitive.
* Hesiod compiles on systems which don't define C_HS or T_TXT.
* Bugs fixed: "hesinfo -b" works, better length-checking on response
  packets, string termination error in configuration file reading,
  README left out periods in sample rhs and lhs values, "make check"
  didn't set fixed hesiod configuration, manual pages weren't
  installed.

Major changes in release 2.0.0:
* The hesiod.3 man page now documents all exported functions.
* Reentrant interfaces were added to the API.  See the man page for
  details.
* hes_strerror() was added to the API.
* Function prototypes were added to <hesiod.h>.  Hesiod now assumes a
  compiler which can handle most ANSI constructs.
* Bugs fixed: memory leaks in hes_getpwnam(), hes_getmailhost,
  hes_getservbyname(); bounds checking added in res_scan().
* Hesiod builds with Autoconf instead of imake now.
* An automated test driver is included.
* The maximum Hesiod response length is no longer coupled to the DNS
  datagram transport size limit.


hesinfo.c
/* This is the source code for the hesinfo program, used to test the
 * Hesiod name server.
 *
 *	$Source: /afs/dev.mit.edu/source/repository/athena/lib/hesiod/hesinfo.c,v $
 *	$Author: treese $
 *	$Athena: hesinfo.c,v 1.1 88/06/05 17:50:15 treese Locked $
 *	$Log: not supported by cvs2svn $
 *
 * Copyright 1988 by the Massachusetts Institute of Technology.  See the
 * file <mit-copyright.h> for copying and distribution information.
 */

#include "mit-copyright.h"

#ifndef lint
static char rcsid_hesinfo_c[] = "$Header: /afs/dev.mit.edu/source/repository/athena/lib/hesiod/hesinfo.c,v 1.2 1988-06-05 19:51:18 treese Exp $";
#endif

#include <stdio.h>

main(argc, argv)
char *argv[];
{
	register char *cp, **cpp;
	char *hes_to_bind(), **hes_resolve();
	int lflag = 0, errflg = 0, bflag = 0;
	extern int optind;
	char *identifier, *type;
	int c;
	
	while ((c = getopt(argc, argv, "lb")) != EOF) {
		if (c == 'l') lflag = 1;
		else if (c == 'b') bflag = 1;
		else errflg++;
	}
	if (argc - optind != 2 || errflg) {
		fprintf(stderr,"Usage: %s [-bl] identifier type\n",argv[0]);
		fprintf(stderr,"	-l selects long format\n");
		fprintf(stderr,"	-b also does hes_to_bind conversion\n");
		exit(2);
	}
	identifier = argv[optind];
	type = argv[optind+1];
		
	if (bflag) {
		if (lflag)
			printf("hes_to_bind(%s, %s) expands to\n", 
			       identifier, type);
		cp = hes_to_bind(identifier, type);
		if (cp == NULL) {
			printf(" error %d\n", hes_error());
			exit(1);
		} 
		printf("%s\n", cp);
		if (lflag) printf("which ");
	}
	if (lflag) 
		printf("resolves to\n");

	cpp = hes_resolve(identifier, type);
	if (cpp == NULL) { 
		if (lflag) printf("nothing\n");
	} else {
		while(*cpp) printf("%s\n", *cpp++);
	}
	if (!cpp) exit(1);
}
